src/sentry/models/integrations/external_actor.py

```diff
@@ -44,7 +45,12 @@ class Meta:
         unique_together = (("organization", "provider", "external_name", "actor"),)
 
     def delete(self, **kwargs):
-        install = self.integration.get_installation(self.organization_id)
+        from sentry.services.hybrid_cloud.integration import integration_service
+
+        integration = integration_service.get_integration(organization_id=self.organization.id)
```

Comment:
How does this service call know which integration to get? Previously the attribute read would use `self.integration_id` as a condition for the relation. Should we also have that condition in the service call?

Reply:
Hmm, good point. As far as I can recall, I was expecting the two methods together (`get_integration` followed by `get_installation`) to resolve to a unique installation, but yeah, there's no reason that the first would return a unique integration. Worrisome that it didn't trip a unit test. I'll investigate further.

Reply:
I bet the test only has a single integration configured :dice:

Reply:
Should this query be using `self.integration_id` instead of the organization id? That would let you do the integration -> org integration query flow you described.

Reply:
https://github.com/getsentry/sentry/pull/48496